web/datapipeline/sles_subsciptions_projector.go

```diff
@@ -0,0 +1,49 @@
+package datapipeline
+
+import (
+	log "github.com/sirupsen/logrus"
+
+	"github.com/trento-project/trento/internal/subscription"
+	"github.com/trento-project/trento/web/entities"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+)
+
+func NewSlesSubscriptionsProjector(db *gorm.DB) *projector {
+	subsProjector := NewProjector("sles_subscriptions", db)
+
+	subsProjector.AddHandler(SubscriptionDiscovery, subsProjector_SubscriptionDiscoveryHandler)
+
+	return subsProjector
+}
+
+func subsProjector_SubscriptionDiscoveryHandler(dataCollectedEvent *DataCollectedEvent, db *gorm.DB) error {
+	decoder := getPayloadDecoder(dataCollectedEvent.Payload)
+
+	var discoveredSubscriptions subscription.Subscriptions
+	if err := decoder.Decode(&discoveredSubscriptions); err != nil {
+		log.Errorf("can't decode data: %s", err)
+		return err
+	}
+
+	var subEntities []entities.SlesSubscription
+
+  for _, subscription := range discoveredSubscriptions {
+    subEntity := entities.SlesSubscription{
+			AgentID:            dataCollectedEvent.AgentID,
+      ID:                 subscription.Identifier,
+    	Version:            subscription.Version,
+    	Arch:               subscription.Arch,
+    	Status:             subscription.Status,
+    	StartsAt:           subscription.StartsAt,
+    	ExpiresAt:          subscription.ExpiresAt,
+    	SubscriptionStatus: subscription.SubscriptionStatus,
+  	}
+
+		subEntities = append(subEntities, subEntity)
+  }
+
+	return db.Clauses(clause.OnConflict{
+		UpdateAll: true,
+	}).Create(&subEntities).Error
+}
```

Comment:
I think we need to delete also the subscription that are not present anymore here,
like just affer creating:
```go
if err := tx.Not(subEntity).Delete(&entities.SlesSubscription{}).Error; err != nil {
		return err
	}
```

Reply:
Hey @fabriziosestito ,
Your suggestion was not working completely. For example in the case where the subscription list was coming empty.
I have implemented it this way: https://github.com/trento-project/trento/pull/461/commits/7ce64e8768409748aa527dfeae1acb20e6615c8a

By the way, this removal thing looks even needed in the clusters and sap systems. It is much more complicated as the cluster belongs to multiple nodes, so an straightforward removal is not an option

Reply:
@arbulu89 you are right, we need to deal with the deletion (or filtering)  in clusters and sap systems as well. I will think about it and get back to you.
Maybe I misunderstood the use-case here, but I thought it could be possible that an agent has an expired subscription and it doesn't send the data regarding that subscription anymore. In that case maybe deleting the row here is the way to go. What are your thoughts?

Reply:
Your assumption is Ok, and not only that, the user can remove their subscription letting the machine unsubscribed (for any reason they have). In this case, we would receive an empty list of subscriptions.
That's why I implemented the `1st delete all, 2nd add the new subscriptions`.
I didn't find any way of doing it, keeping in mind that we cannot remove all the subscriptions, we need to take care of the subscription of this particular host.